bundles/framework/myplaces3/CategoryHandler.js

```diff
@@ -181,15 +181,21 @@ Oskari.clazz.define('Oskari.mapframework.bundle.myplaces3.CategoryHandler',
             this._notifyUpdate();
         },
         editCategory: function (categoryId) {
-            const layer = this.getCategory(categoryId);
-            const saveLayer = (name, style) => {
+            const layer = this.mapLayerService.findMapLayer(this.getMapLayerId(categoryId));
+            const saveLayer = (locale, style) => {
                 this.saveCategory({
-                    ...layer,
-                    name,
+                    id: categoryId,
+                    locale,
                     style
                 });
             };
-            showModal(layer.name, layer.style, saveLayer);
+            const names = { ...layer._name }; // TODO: getter?
```

Comment:
Lets add a getter for "raw" name or locale or something other than using `layer._name` directly.